### PR TITLE
feat(cast): adds support for gas, value, nonce for send cmd & value for estimate cmd

### PR DESCRIPTION
**Pull request by Janmajayamall**
## Motivation

1.  Ability to supply gas amount, value, nonce with cast send cmd and value with cast estimate cmd  allows for greater flexibility when dealing with transactions.
2. Ability to specify ETH_KEYSTORE (like dapp) for wallet use allows one to avoid specifying keystore file every time.

## Solution

1. Added **gas**, **value**, and **nonce** options to **cast send** cmd.
For example -
`
cast send 0xa6B71E26C5e0845f74c812102Ca7114b6a896AB2 "createProxyWithNonce(address,bytes,uint256)" 0xd9Db270c1B5E3Bd161E8c8503c55cEABeE709552 "" 13 --gas 250000 --value 1 --nonce 7 --private-key PV_KEY
`
2. Added **value** options to **cast estimate** cmd.
For example -
`
cast send 0xa6B71E26C5e0845f74c812102Ca7114b6a896AB2 "createProxyWithNonce(address,bytes,uint256)" 0xd9Db270c1B5E3Bd161E8c8503c55cEABeE709552 "" 13 --value 1
`
3. Added support for **ETH_KEYSTORE** env variable for specifying keystore file for wallet
For example, now can use wallet by specifying **ETH_KEYSTORE** in env like this  - 
`
export ETH_RPC_URL=PATH_TO_KEYSTORE_FILE
`

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -84,7 +84,7 @@ Please also make sure that the following commands pass if you have changed the c
 cargo check --all
 cargo test --all --all-features
 cargo +nightly fmt -- --check
-cargo +nightly clippy --all --all-features -- -D warnings
+cargo +nightly clippy -all --all-features -- -D warnings
```
Comment (mattsse) [inline CONTRIBUTING.md:87]:
```suggestion
cargo +nightly clippy --all --all-features -- -D warnings
```

File: cli/src/opts/cast.rs [hunk 2]
```diff
@@ -147,6 +147,12 @@ pub enum Subcommands {
         sig: String,
         #[clap(help = "the list of arguments you want to call the function with")]
         args: Vec<String>,
+        #[clap(long, help = "gas quantity for the transaction")]
+        gas: Option<U256>,
+        #[clap(long, help = "ether value for the transaction")]
```
Comment (mattsse) [inline cli/src/opts/cast.rs:152]: needs more clarification regarding unit

**Review by onbjerg — CHANGES_REQUESTED** [review]

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit,
otherwise looks correct.
The functions are a bit verbose now, but that's fine for now

File: CONTRIBUTING.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (Janmajayamall) [reply CONTRIBUTING.md]: yup done! :)

Comment (Janmajayamall) [reply CONTRIBUTING.md]: I rebased to latest master

File: cli/src/opts/cast.rs (hunk 2 above)
Comment (Janmajayamall) [reply cli/src/opts/cast.rs:152]: would - **ether value (in wei) for the transaction** make it clear?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -84,7 +84,7 @@ Please also make sure that the following commands pass if you have changed the c
 cargo check --all
 cargo test --all --all-features
 cargo +nightly fmt -- --check
-cargo +nightly clippy --all --all-features -- -D warnings
+cargo +nightly clippy -al --all-features -- -D warnings
```
Comment (gakonst) [inline CONTRIBUTING.md:87]:
```suggestion
cargo +nightly clippy --all --all-features -- -D warnings
```

File: cli/src/opts/cast.rs (hunk 2 above)
Comment (gakonst) [inline cli/src/opts/cast.rs:152]:
```suggestion
        #[clap(long, help = "ether value (in wei) for the transaction")]
```

File: cli/src/opts/cast.rs [hunk 4]
```diff
@@ -171,6 +177,8 @@ pub enum Subcommands {
         sig: String,
         #[clap(help = "the list of arguments you want to call the function with")]
         args: Vec<String>,
+        #[clap(long, help = "value for tx estimate")]
```
Comment (gakonst) [inline cli/src/opts/cast.rs:180]:
```suggestion
        #[clap(long, help = "value for tx estimate (in wei)")]
```

**Review by gakonst — APPROVED** [review]

**Pull request merged**